api/tests/unit/infrastructure/repositories/answer-repository_test.js

```diff
@@ -0,0 +1,117 @@
+const { describe, it, before, after, expect, knex } = require('../../../test-helper');
+
+const AnswerRepository = require('../../../../lib/infrastructure/repositories/answer-repository');
+
+describe('Unit | Repository | AnswerRepository', function () {
+
+
+  describe('findByChallengeAndAssessment', function () {
+
+    // nominal case
+    const inserted_answer_1 = {
+      value: '1,2',
+      result: 'ko',
+      challengeId: 'challenge_1234',
+      assessmentId: 1234
+    };
+
+    // same assessmentId, different challengeId
+    const inserted_answer_2 = {
+      value: '1,2,4',
+      result: 'ok',
+      challengeId: 'challenge_000',
+      assessmentId: 1234
+    };
+
+    // different assessmentId, same challengeId
+    const inserted_answer_3 = {
+      value: '3',
+      result: 'partially',
+      challengeId: 'challenge_1234',
+      assessmentId: 1
+    };
+
+    before(function (done) {
+      knex('answers').delete().then(() => {
+        knex('answers').insert([inserted_answer_1, inserted_answer_2, inserted_answer_3]).then(() => {
+          done();
+        });
+      });
+    });
+
+    after(function (done) {
+      knex('answers').delete().then(() => {done();});
+    });
+
+    it('should findByChallengeAndAssessment', function (done) {
```

Comment:
changé en "should find the answer by challenge and assessment and return its in an object"